src/internal/pachsql/table_info.go

```diff
@@ -0,0 +1,82 @@
+package pachsql
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"strings"
+
+	"github.com/pachyderm/pachyderm/v2/src/internal/errors"
+)
+
+// TableInfo contains information about a SQL table
+type TableInfo struct {
+	Columns []ColumnInfo
+}
+
+// ColumnInfo is information about a single column in a SQL table
+type ColumnInfo struct {
+	TableSchema string
+	TableName   string
+	DataType    string
+	IsNullable  bool
+}
+
+// GetTableInfo looks up information about the table using INFORMATION_SCHEMA
+func GetTableInfo(ctx context.Context, db *DB, tableName string) (*TableInfo, error) {
+	readonly := true
+	if db.DriverName() == "snowflake" {
+		readonly = false
+	}
+	tx, err := db.BeginTxx(ctx, &sql.TxOptions{ReadOnly: readonly}) // TODO Snowflake doesn't support ReadOnly
+	if err != nil {
+		return nil, errors.EnsureStack(err)
+	}
+	defer tx.Rollback()
+	ti, err := GetTableInfoTx(tx, tableName)
+	if err != nil {
+		return nil, errors.EnsureStack(err)
+	}
+	return ti, errors.EnsureStack(tx.Rollback())
+}
+
+// GetTableInfoTx looks up information about the table using INFORMATION_SCHEMA
+func GetTableInfoTx(tx *Tx, tablePath string) (*TableInfo, error) {
+	schemaName, tableName := SplitTableSchema(tx.DriverName(), tablePath)
+	var cinfos []ColumnInfo
+	q := `SELECT table_schema, table_name, data_type, is_nullable
+		FROM INFORMATION_SCHEMA.columns
+	`
+	var args []interface{}
+	// table_name
+	q += fmt.Sprintf(" WHERE lower(table_name) = lower(%s)", Placeholder(tx.DriverName(), len(args)))
+	args = append(args, tableName)
+	// schema_name
+	if schemaName != "" {
```

Comment:
In MySQL, the user won't provide a schema name since that concept doesn't really exist. The schema name is the same as the database name in MySQL. Since the database name is part of the connection string, this query would be executed within that context.

Reply:
Ah I see, in db's where there is the concept of a schema, there could be two tables of the same name in different schemas, and we could get column info from both tables without checking for schema.

I think there are two ways to handle this:
1. For the user to supply us a schema for systems that do have it
2. Just assume that the user has setup their database correctly, and that if they don't supply schema, then the table name is unique.

Reply:
We could also error if we see columns from another schema. I'd like to do something about this case - preventing it or erroring here would be a lot clearer than erroring when putting the actual data in, I think